src/esockd_listener.erl

```diff
@@ -118,6 +134,20 @@ handle_call(get_port, _From, State = #state{lport = LPort}) ->
 handle_call(get_lsock, _From, State = #state{lsock = LSock}) ->
     {reply, LSock, State};
 
+handle_call(get_state, _From, State = #state{lsock = LSock, lport = LPort}) ->
+    Reply = [ {listen_sock, LSock}
+            , {listen_port, LPort}
+            ],
+    {reply, Reply, State};
+
+handle_call({set_options, Opts}, _From, State = #state{lsock = LSock}) ->
```

Comment:
should maybe merge the new Opts instead ? i.e. make it incremental.

Reply:
I believe `inet:setopts/2` only updates listed options, and not replaces all of them.

> should maybe merge the new Opts instead ? i.e. make it incremental.

Good point. Though I suppose it probably should happen on `esockd:merge_opts/2` level. Otherwise, some of these changes may be lost during listener restart for example.

> also, otherwise DEFAULT_TCP_OPTIONS is lost ? since it's not merged in sockopts/1

If my understanding of `inet:setopts/2` is correct, it should be fine to set them only during `init/1`.

Reply:
I feel it's better to only call inet:setopts with the provided (incremental) opts in this gen_call.
And then merge the new opts to the options in the state record.

It should not be the caller's responsibility to get, merge then call this API.

Reply:
> It should not be the caller's responsibility to get, merge then call this API.

This is now taken care of by `esockd_listener_sup:set_options/3` as of 84e0b78. In effect, `inet:setopts/2` is getting called with a merged set of options.

Reply:
Yes, I understand it's merged before the gen_call.

My fear is that some day, I might forget about it and evaluate gen_server:call(Pid, ...) with only a few opts from a console.

Reply:
I see. But `inet:setopts/2` should be incremental by itself anyway, shouldn't it? Also made 4dbb63a to ensure that there's no inconsistent gen_server state in this case.